File: z-clients/javascript/src/models/svixPollerOut.ts
// this file is @generated

export interface SvixPollerOut {
    topic: string;
    pollerId: string;
    /** The autoconfig token, obfuscated (e.g. `auto_v1_eyJh...fQ==`). */
    token: string;
}

export const SvixPollerOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): SvixPollerOut {
        return {
            topic: object['topic'],
            pollerId: object['poller_id'],
            token: object['token'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: SvixPollerOut): any {
        return {
            'topic': self.topic,
            'poller_id': self.pollerId,
            'token': self.token,
        };
    }
}

File: z-clients/rust/src/models/svix_poller_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct SvixPollerOut {
    pub topic: String,

    pub poller_id: String,

    /// The autoconfig token, obfuscated (e.g. `auto_v1_eyJh...fQ==`).
    pub token: String,
}

impl SvixPollerOut {
    pub fn new(topic: String, poller_id: String, token: String) -> Self {
        Self {
            topic,
            poller_id,
            token,
        }
    }
}
